packages/next/src/build/transpile-config.ts

```diff
@@ -0,0 +1,49 @@
+import { join } from 'path'
+import { readFile, writeFile } from 'fs/promises'
+import { transform } from './swc'
+
+type Log = {
+  error: (message: string) => void
+}
+
+export async function transpileConfig({
+  configPath,
+  configFileName,
+  cwd,
+  log,
+}: {
+  configPath: string
+  configFileName: string
+  cwd: string
+  log: Log
+}): Promise<string> {
+  const isCJS = configFileName.endsWith('cts')
+  const compiledConfigPath = join(
+    cwd,
+    // '.next',
+    `next.compiled.config.${isCJS ? 'js' : 'mjs'}`
```

Comment:
If the developer adds `module: true` to their root `package.json`, the compiled `.js` file will still be parsed as an ESM module. So this is still wrong.

Reply:
AFAIK next config doesn't recommend that case but suggests to use `mjs` extension for ESM